packages/flutter_tools/lib/src/test/flutter_web_platform.dart

```diff
@@ -167,6 +193,41 @@ class FlutterWebPlatform extends PlatformPlugin {
     }
   }
 
+  final bool updateGoldens;
+  _TestGoldenComparator _testGoldenComparator;
+
+  Future<shelf.Response> _goldenFileHandler(shelf.Request request) async {
+    if (request.url.path.contains('flutter_goldens')) {
+      final Map<String, Object> body = json.decode(await request.readAsString());
+      final Uri goldenKey = Uri.parse(body['key']);
+      final Uri testUri = Uri.parse(body['testUri']);
+      final num width = body['width'];
+      final num height = body['height'];
+      final Runtime browser = Runtime.chrome;
+      final BrowserManager browserManager = await _browserManagerFor(browser);
+      final ChromeTab chromeTab = await browserManager._browser.chromeConnection.getTab((ChromeTab tab) {
```

Comment:
This code should handle the errors thrown by WIP, like WIPError (there might be some more). probably printError and move on if possible

Reply:
Done, I've handled it by catching WIPError and returning the error to the caller.